tensorflow_privacy/privacy/membership_inference_attack/seq2seq_mia_test.py

```diff
@@ -0,0 +1,355 @@
+                     dtype=object)
+        ]),
+        logits_test=iter([
+            np.array([
+                np.array([[0.25, 0.4, 0.35], [0.2, 0.4, 0.4]], dtype=np.float32)
+            ],
+                     dtype=object),
+            np.array([
+                np.array([[0.3, 0.3, 0.4], [0.4, 0.4, 0.2]], dtype=np.float32),
+                np.array([[0.3, 0.35, 0.35]], dtype=np.float32)
+            ],
+                     dtype=object),
+            np.array([
+                np.array([[0.25, 0.4, 0.35], [0.2, 0.4, 0.4]], dtype=np.float32)
+            ],
+                     dtype=object)
+        ]),
+        labels_train=iter([
+            np.array([
+                np.array([2, 0], dtype=np.float32),
+                np.array([1], dtype=np.float32)
+            ],
+                     dtype=object),
+            np.array([np.array([1, 0], dtype=np.float32)], dtype=object),
+            np.array([
+                np.array([0, 1], dtype=np.float32),
+                np.array([1, 2], dtype=np.float32)
+            ],
+                     dtype=object)
+        ]),
+        labels_test=iter([
+            np.array([np.array([2, 1], dtype=np.float32)]),
+            np.array([
+                np.array([2, 0], dtype=np.float32),
+                np.array([1], dtype=np.float32)
+            ],
+                     dtype=object),
+            np.array([np.array([2, 1], dtype=np.float32)])
+        ]),
+        vocab_size=3,
+        train_size=3,
+        test_size=3)
+    privacy_report_metadata = PrivacyReportMetadata()
+    attacker_data = create_seq2seq_attacker_data(attack_input_data=attack_input,
+                                                 test_fraction=0.33,
+                                                 balance=True,
+                                                 privacy_report_metadata=privacy_report_metadata)
+    self.assertLen(attacker_data.features_train, 4)
+    self.assertLen(attacker_data.features_test, 2)
+
+    for _, feature in enumerate(attacker_data.features_train):
+      self.assertLen(feature, 1)  # each feature has one average rank
+
+    # Tests that fields of PrivacyReportMetadata are populated.
+    self.assertIsNotNone(privacy_report_metadata.loss_train)
+    self.assertIsNotNone(privacy_report_metadata.loss_test)
+    self.assertIsNotNone(privacy_report_metadata.accuracy_train)
+    self.assertIsNotNone(privacy_report_metadata.accuracy_test)
+
+
+def _get_batch_logits_and_labels(num_sequences, max_tokens_in_sequence,
+                                 vocab_size):
+  num_tokens_in_sequence = np.random.choice(max_tokens_in_sequence,
+                                            num_sequences) + 1
+  batch_logits, batch_labels = [], []
+  for num_tokens in num_tokens_in_sequence:
+    logits, labels = _get_sequence_logits_and_labels(num_tokens, vocab_size)
+    batch_logits.append(logits)
+    batch_labels.append(labels)
+  return np.array(
+      batch_logits, dtype=object), np.array(
```

Comment:
This was done because numpy was throwing a warning about creating a 2D array with variable length arrays inside it (because a batch will have sequences of diff lengths)